src/main.rs

```diff
@@ -113,16 +116,35 @@ mod filters {
             .and(warp::query::<GetQuery>())
             .and(with_db(db))
             .and_then(move |query: GetQuery, db: Database| async move {
-                let filter_options = mongodb::options::FindOneOptions::builder().build();
-                let filter = bson::doc! { "path": query.path };
 
-                match db.collection("data").find_one(filter, filter_options).await {
-                    Ok(Some(doc)) => {
-                        let data: Data = bson::from_document(doc).unwrap();
-                        Ok(warp::reply::json(&data))
+                if let Some(_) = query.skip {
+                    // If there is a skip query param, retrieve collection
+
+                    let filter_options = mongodb::options::FindOptions::builder().skip(query.skip).limit(query.page_size).build();
+                    let filter = bson::doc! { "path": query.path };
+
+                    match db.collection("data").find(filter, filter_options).await {
+                        Ok(mut cursor) => {
+                            let mut results_vector = Vec::new();
+                            while let Some(item) = cursor.next().await {
+                                results_vector.push(item.unwrap())
+                            }
+                            Ok(warp::reply::json(&results_vector))
+                        },
+                        Err(e) => Err(warp::reject::reject()),
```

Comment:
Not for addressing in this PR, but we need to switch all these generic rejects into proper error-handling. I did the exact same thing in the FindOne route cuz I didn't know how to do proper warp error-handling, but let's make a separate PR that closes up these gaps.